new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/IstioUtils.java

```diff
@@ -0,0 +1,271 @@
+       String domainNamespace, String domainUid, 
+       String adminServiceName, String clusterServiceName) throws IOException {
+
+    logger.info("Create a staging location for istio configuration objects");
+    Path fileTemp = Paths.get(RESULTS_ROOT, "tmp", "istio");
+    deleteDirectory(fileTemp.toFile());
+    Files.createDirectories(fileTemp);
+
+    logger.info("Copy istio-http-template.service.yaml to staging location");
+    Path srcFile = Paths.get(RESOURCE_DIR, "istio", "istio-http-template.service.yaml");
+    Path targetFile = Paths.get(fileTemp.toString(), "istio-http-service.yaml");
+    Files.copy(srcFile, targetFile, StandardCopyOption.REPLACE_EXISTING);
+
+    String out = targetFile.toString();
+    replaceStringInFile(out, "NAMESPACE", domainNamespace); 
+    replaceStringInFile(out, "DUID", domainUid); 
+    replaceStringInFile(out, "ADMIN_SERVICE", adminServiceName); 
+    replaceStringInFile(out, "CLUSTER_SERVICE", clusterServiceName); 
+    ExecResult result = null;
+    StringBuffer deployIstioGateway = null;
+    deployIstioGateway = new StringBuffer("kubectl apply -f ");
+    deployIstioGateway.append(targetFile);
+    logger.info("deployIstioGateway: kubectl command {0}", new String(deployIstioGateway));
+    try {
+      result = exec(new String(deployIstioGateway), true);
+    } catch (Exception ex) {
+      logger.info("Exception in deployIstioGateway() {0}", ex);
+      return false;
+    }
+    logger.info("deployIstioHttpGateway: kubectl returned {0}", result.toString());
+    if (result.stdout().contains("istio-http-gateway created")) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+  
+  /**
+   * Deploy the tcp Istio Gateway and Istio Virtualservice.
+   * @param domainNamespace name of the domain namespace
+   * @param domainUid name of the domain identifier
+   * @param adminServiceName name of the admin service
+   * @returns true if deployment is success otherwise false
+  */
+  public static boolean deployTcpIstioGatewayAndVirtualservice(
+        String domainNamespace, String domainUid, String adminServiceName) throws IOException  {
+
+    logger.info("Create a staging location for istio configuration objects");
+    Path fileTemp = Paths.get(RESULTS_ROOT, "tmp", "istio");
+    deleteDirectory(fileTemp.toFile());
+    Files.createDirectories(fileTemp);
+
+    logger.info("Copy istio-tcp-template.service.yaml to staging location");
+    Path srcFile = Paths.get(RESOURCE_DIR, "istio", "istio-tcp-template.service.yaml");
+    Path targetFile = Paths.get(fileTemp.toString(), "istio-tcp-service.yaml");
+    Files.copy(srcFile, targetFile, StandardCopyOption.REPLACE_EXISTING);
+
+    String out = targetFile.toString();
+    String adminService = adminServiceName + ".svc.cluster.local";
+    replaceStringInFile(out, "NAMESPACE", domainNamespace); 
+    replaceStringInFile(out, "DUID", domainUid); 
+    replaceStringInFile(out, "ADMIN_SERVICE", adminServiceName); 
```

Comment:
Same comment as for the Http service above.

Reply:
done